gate-core/src/main/java/com/netflix/spinnaker/gate/services/PermissionService.java

```diff
@@ -133,22 +136,27 @@ public Set<Role.View> getRoles(String userId) {
       var permission = permissionEvaluator.getPermission(userId);
       var roles = permission != null ? permission.getRoles() : null;
       return roles != null ? roles : Set.of();
-    } catch (RetrofitError e) {
+    } catch (SpinnakerServerException e) {
       throw UpstreamBadRequest.classifyError(e);
     }
   }
 
   List<UserPermission.View> lookupServiceAccounts(String userId) {
     try {
       return extendedFiatService.getUserServiceAccounts(userId);
-    } catch (RetrofitError re) {
-      var response = re.getResponse();
-      if (response != null && response.getStatus() == HttpStatus.NOT_FOUND.value()) {
+    } catch (SpinnakerNetworkException e) {
+      throw new SystemException("getUserServiceAccounts failed", e).setRetryable(true);
+    } catch (SpinnakerHttpException e) {
+      if (e.getResponseCode() == 404) {
         return List.of();
       }
-      boolean shouldRetry =
-          response == null || HttpStatus.valueOf(response.getStatus()).is5xxServerError();
-      throw new SystemException("getUserServiceAccounts failed", re).setRetryable(shouldRetry);
+      if (HttpStatus.valueOf(e.getResponseCode()).is5xxServerError()) {
+        throw new SystemException("getUserServiceAccounts failed", e).setRetryable(true);
+      }
+      throw UpstreamBadRequest.classifyError(e);
+    } catch (SpinnakerServerException e) {
+      e.setRetryable(false);
```

Comment:
No need for this right?  UpstreamBadRequest handles retryable, and `e` doesn't live beyond this catch block.

Reply:
I thought so but without setting here, retryable field is set to null causing tests to fail.

Reply:
It can be set in UpstreamBadRequest as well.

Reply:
But...doesn't the RetrofitError-based code have null for retryable?

Reply:
No, retryable value was set for all the cases for RetrofirError-based code in this line :

```
throw new SystemException("getUserServiceAccounts failed", re).setRetryable(shouldRetry);
```

Reply:
Turns out that [SpinnakerConversionException](https://github.com/spinnaker/kork/blob/cf76e731d4201d0972cec6197a0f2536046e4537/kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerConversionException.java) constructors always set retryable to false, so there's no need to do it here.